connectors/tests/test_byoc.py

```diff
@@ -956,56 +956,105 @@ def test_nested_get(nested_dict, keys, default, expected):
     assert expected == Features(nested_dict)._nested_feature_enabled(keys, default)
 
 
-def test_pending_job_query_with_connectors_ids(mock_responses, set_env):
+JOB_SOURCE = {"_id": "1", "_source": {"status": "pending", "connector": {"id": "1"}}}
+
+
+class AsyncIterator:
+    def __init__(self, seq):
+        self.iter = iter(seq)
+
+    def __aiter__(self):
+        return self
+
+    async def __anext__(self):
+        try:
+            return next(self.iter)
+        except StopIteration:
+            raise StopAsyncIteration
```

Comment:
This is an async wrapper of iterator, to patch async generator function. Not sure whether there's any better way to do so.

This is also imported in `test_job_cleanup.py`. Is there any shared test support file for it?

cc. @tarekziade

Reply:
we can add a `connectors/tests/support.py` to share stuff like that